package-res/resources/web/pentaho/lang/Base.js

```diff
@@ -83,6 +84,8 @@ define([
         "extend_exclude": 1
       });
 
+  var _isDebugMode = debugLevel.level > 4; // 5 is "debug"
```

Comment:
`debugLevel` will ultimately parse the url for  `debug=` and `debugLevel=` parameters. 
While this is the current practice, I'm twitching to the sight of this: I don't like having`pentaho.lang.Base` parse the location url, even if indirectly.

I understand that this is a practical way for a developer to globally define the debug flags, but I need some time  to overcome my prejudice.